android/java/org/chromium/chrome/browser/bookmarks/BraveBookmarkManager.java

```diff
@@ -0,0 +1,151 @@
+
+    public BraveBookmarkManager(Context context, ComponentName openBookmarkComponentName,
+            boolean isDialogUi, boolean isIncognito, SnackbarManager snackbarManager) {
+        super(context, openBookmarkComponentName, isDialogUi, isIncognito, snackbarManager);
+    }
+
+    public void setWindow(ActivityWindowAndroid window) {
+        mWindowAndroid = window;
+    }
+
+    @Override
+    public void importBookmarks() {
+        if (mWindowAndroid == null || mContext == null) {
+            return;
+        }
+
+        if (DeviceFormFactor.isNonMultiDisplayContextOnTablet(mContext)) {
+            doImportBookmarks();
+        } else {
+            if (mWindowAndroid.hasPermission(Manifest.permission.READ_EXTERNAL_STORAGE)) {
+                doImportBookmarks();
+            } else {
+                mWindowAndroid.requestPermissions(
+                        new String[] {Manifest.permission.READ_EXTERNAL_STORAGE},
+                        (permissions, grantResults) -> {
+                            if (grantResults.length >= 1
+                                    && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                                doImportBookmarks();
+                            }
+                        });
+            }
+        }
+    }
+
+    private void doImportBookmarks() {
+        Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.setType("text/html");
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+        if (mWindowAndroid.showIntent(
+                    Intent.createChooser(intent, "R.string.import_bookmarks_select_file"),
+                    new WindowAndroid.IntentCallback() {
+                        @Override
+                        public void onIntentCompleted(int resultCode, Intent results) {
+                            if (resultCode == Activity.RESULT_OK && results != null
+                                    && results.getData() != null) {
+                                try {
+                                    Cursor cursor = mContext.getContentResolver().query(
+                                            results.getData(), null, null, null, null);
+                                    int nameIndex =
+                                            cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME);
+                                    cursor.moveToFirst();
+                                    String name = cursor.getString(nameIndex);
+                                    File file = new File(mContext.getFilesDir(), name);
+                                    InputStream inputStream =
+                                            mContext.getContentResolver().openInputStream(
+                                                    results.getData());
+                                    FileOutputStream outputStream = new FileOutputStream(file);
+                                    int read = 0;
```

Comment:
Maybe it is enough to have only one variable instead of both `int read = 0;` and `int k`?
`int bytesRead`?